mobile/src/main/java/mycroft/ai/MainActivity.java

```diff
@@ -452,8 +459,30 @@ protected void checkIfLaunchedFromWidget(Intent intent) {
         }
     }
 
+    private void loadVersionInfo() {
+        String versionName = "";
+        int versionCode = -1;
+        try {
+            PackageInfo packageInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
+            versionName = packageInfo.versionName;
+            versionCode = packageInfo.versionCode;
+
+            SharedPreferences.Editor editor = sharedPref.edit();
+            editor.putInt("versionCode", versionCode);
+            editor.putString("versionName", versionName);
+            editor.commit();
```

Comment:
there's a quick writeup on the difference at http://stackoverflow.com/a/5960732/3934789